### PR TITLE
Add keccak native support

### PR DESCRIPTION
**Pull request by k06a**
Related to: https://github.com/nearprotocol/near-bindgen/pull/69

### CODE REVIEW

**Comment by evgenykuzyakov** [comment]
Seems you need to run `cargo rust fmt`.

File: runtime/near-vm-logic/src/logic.rs [hunk 1]
```diff
@@ -739,6 +739,42 @@ impl<'a> VMLogic<'a> {
...
+    /// Hashes the random sequence of bytes using keccak512 and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + sha256_base + sha256_byte * num_bytes`
```
Comment (evgenykuzyakov) [inline runtime/near-vm-logic/src/logic.rs]: update this comment as well.

File: runtime/near-vm-logic/src/logic.rs [hunk 2]
```diff
@@ -739,6 +739,42 @@ impl<'a> VMLogic<'a> {
...
+    /// Hashes the random sequence of bytes using keccak256 and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + sha256_base + sha256_byte * num_bytes`
```
Comment (evgenykuzyakov) [inline runtime/near-vm-logic/src/logic.rs]: update this comment.

**Comment by evgenykuzyakov** [comment]
Please also update `nomicon` https://nomicon.io/Runtime/Components/BindingsSpec/MathAPI.html

File: runtime/near-vm-logic/src/logic.rs (hunk 2 above)
Comment (k06a) [reply runtime/near-vm-logic/src/logic.rs]: Done

File: runtime/near-vm-logic/src/logic.rs (hunk 1 above)
Comment (k06a) [reply runtime/near-vm-logic/src/logic.rs]: Done

**Comment by evgenykuzyakov** [comment]
@k06a I got an error when trying to launch local testnet with this branch:
```
Starting NEAR client...
Autoupdate is not supported at the moment for runs outside of docker
thread 'main' panicked at 'Failed to deserialize the genesis config.: Error("missing field `keccak256_base`", line: 173, column: 7)', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Not sure why it's needed in genesis. I thought we should be just using defaults. Something went wrong I guess

**Comment by k06a** [comment]
@evgenykuzyakov try to erase `./testdir` in repo

**Pull request merged**